x/stake/keeper/validator.go

```diff
@@ -157,28 +132,22 @@ func (k Keeper) GetValidatorsBonded(ctx sdk.Context) (validators []Validator) {
 }
 
 // get the group of bonded validators sorted by power-rank
-func (k Keeper) GetValidatorsByPower(ctx sdk.Context) []Validator {
+func (k Keeper) GetValidatorsByPower(ctx sdk.Context) []types.Validator {
 	store := ctx.KVStore(k.storeKey)
 	maxValidators := k.GetParams(ctx).MaxValidators
-	validators := make([]Validator, maxValidators)
-	iterator := sdk.KVStoreReversePrefixIterator(store, ValidatorsByPowerKey) // largest to smallest
+	validators := make([]types.Validator, maxValidators)
+	iterator := sdk.KVStoreReversePrefixIterator(store, ValidatorsByPowerIndexKey) // largest to smallest
 	i := 0
 	for {
 		if !iterator.Valid() || i > int(maxValidators-1) {
 			iterator.Close()
```

Comment:
Can this be closed outside of the loop, per #1125

Reply:
addressed